src/gui/sdlmain.cpp

```diff
@@ -3842,20 +3817,42 @@ bool GFX_Events()
 				focus_input();
 				continue;
 
-			case SDL_WINDOWEVENT_RESIZED:
+			case SDL_WINDOWEVENT_RESIZED: {
 				// TODO pixels or logical unit?
-				// LOG_DEBUG("SDL: Window has been resized to %dx%d",
-				//               event.window.data1,
-				//               event.window.data2);
-				//
+				// LOG_DEBUG("SDL: Window has been resized to
+				// %dx%d",
+				//           event.window.data1,
+				//           event.window.data2);
+
 				// When going from an initial fullscreen to
 				// windowed state, this event will be called
 				// moments before SDL's windowed mode is
 				// engaged, so simply ensure the window size has
 				// already been established:
 				assert(sdl.desktop.window.width > 0 &&
 				       sdl.desktop.window.height > 0);
-				continue;
+
+				// SDL_WINDOWEVENT_RESIZED events are sent twice on macOS when
+				// resizing the window, so we're only logging the display
+				// settings if there is a change since the last window resized
+				// event.
+				static int prev_width  = 0;
+				static int prev_height = 0;
+
+				const auto new_width  = event.window.data1;
+				const auto new_height = event.window.data2;
+
+				if (prev_width != new_width ||
+				    prev_height != new_height) {
+					log_display_properties(sdl.draw.render_width_px,
+					                       sdl.draw.render_height_px,
+					                       sdl.video_mode,
+					                       sdl.rendering_backend);
+				}
```

Comment:
If we switch VESA color depths, say 15-bit -> 16-bit -> 24-bit, then the mode will change, but the resolution will stay the same.

(And there are probably similar CGA -> EGA -> VGA switches where resolution can be similar?)

Probably safest just to hold and compare all four inputs.

Reply:
No man, this is just the OS level width & height 😄 Just what the windowing system sends, so the new window dimensions 😄

Reply:
Ah! Thanks for clarifying :-)

Reply:
No problem :)